### PR TITLE
adding pytorch lambdify functionality

### PR DESCRIPTION
**Pull request by clayton-r**
#### References to other Issues or PRs
re-attempt at #15484 and #20065 and #20516 (lol) now that pytorch is probably installable in gh actions

#### Brief description of what is fixed or changed
Added PyTorch backend support to lambdify by creating a new printing.pytorch module, modifying utilities.lambdify to support the PyTorch backend, and adding appropriate tests in printing.test.test_torch.

#### Other comments
Unlike sympytorch (https://github.com/patrick-kidger/sympytorch), this implementation does not create trainable PyTorch modules. It seems that there is utility in this anyway since sometimes you just need the gradients to flow.

#### Release Notes

* printing
  * Added new PyTorch printer module for lambdify
* utilities
  * Added PyTorch backend support to lambdify

### CODE REVIEW

**Comment by clayton-r** [comment]
For the sake of anyone looking at this PR and wondering about the reference to Sympytorch, see this previous thread:
             
              I think it would make sense. It seems to me like converting to a `torch.nn.Module` is the natural way to "pytorchify" a function. That's not what this PR does though, as mentioned above.

_Originally posted by @patrick-kidger in https://github.com/sympy/sympy/issues/20516#issuecomment-788404078_

**Comment by oscarbenjamin** [comment]
Looks like the torch tests don't run because the script is missing the executable bit.

**Comment by oscarbenjamin** [comment]
This broadly looks good to me although I don't know the lambdify code so well.

Perhaps @bjodah could give this a quick look to see if anything there are any obvious improvements needed?

**Review by bjodah — CHANGES_REQUESTED** [review]
I think this looks good! Third time's the charm, right?

Just a minor nit on a commented block left in. (if it should stay, we should add more commentary explaining why).

**Review by bjodah — APPROVED** [review]

**Comment by bjodah** [comment]
Thank you @clayton-r for seeing this thru!

**Comment by oscarbenjamin** [comment]
I just closed the other PRs. Was there an issue as well or was it just PRs?

**Comment by bjodah** [comment]
Ah, sorry, I should have done that. I don't remember ever seeing a tracking issue.
